docs/admin/out-of-resource.md

```diff
@@ -0,0 +1,220 @@
+| MemoryPressure | memory.available | Available memory on the node has satisfied an eviction threshold |
+
+The `kubelet` will continue to report node status updates at the frequency specified by
+`--node-status-update-frequency` which defaults to `10s`.
+
+### Oscillation of node conditions
+
+If a node is oscillating above and below a soft eviction threshold, but not exceeding
+its associated grace period, it would cause the corresponding node condition to
+constantly oscillate between true and false, and could cause poor scheduling decisions
+as a consequence.
+
+To protect against this oscillation, the following flag is defined to control how
+long the `kubelet` must wait before transitioning out of a pressure condition.
+
+```
+--eviction-pressure-transition-period=5m0s: Duration for which the kubelet has to wait
+before transitioning out of an eviction pressure condition.
+```
+
+The `kubelet` would ensure that it has not observed an eviction threshold being met
+for the specified pressure condition for the period specified before toggling the
+condition back to `false`.
+
+### Eviction of Pods
+
+If an eviction threshold has been met, the `kubelet` will initiate the
+process of evicting pods until it has observed the signal has gone below
+its defined threshold.
+
+The `kubelet` ranks pods for eviction that have the lowest quality of
+service that are the largest consumers of the starved compute resource
+relative to the pods scheduling request.
+
+* `BestEffort` pods that consume the most of the starved resource are failed
+first.
+* `Burstable` pods that consume the greatest amount of the starved resource
+relative to their request for that resource are killed first.  If no pod
+has exceeded its request, the strategy targets the largest consumer of the
+starved resource.
+* `Guaranteed` pods that consume the greatest amount of the starved resource
+relative to their request are killed first.  If no pod has exceeded its request,
+the strategy targets the largest consumer of the starved resource.
+
+A `Guaranteed` pod is guaranteed to never be evicted because of another pod's
+resource consumption.  If a system daemon (i.e. `kubelet`, `docker`, `journald`, etc.)
+is consuming more resources than were reserved via `system-reserved` or `kube-reserved` allocations,
+and the node only has `Guaranteed` pod(s) remaining, then the node must choose to evict a
+`Guaranteed` pod in order to preserve node stability, and to limit the impact
+of the unexpected consumption to other `Guaranteed` pod(s).
+
+### Scheduler
+
+The node will report a condition when a compute resource is under pressure.  The
+scheduler views that condition as a signal to dissuade placing additional
+pods on the node.
+
+| NodeCondition    | Scheduler Behavior                               |
+| ---------------- | ------------------------------------------------ |
+| MemoryPressure | BestEffort pods are not scheduled to the node. |
```

Comment:
How about something around the lines:

```
| MemoryPressurce | If there are no strictly lower quality-of-service pods on that node, a new pod is not scheduled there |
```

That would allow that a node full of BestEffort pods can be re-used for Guaranteed pods.